sdk/src/main/java/com/mapbox/maps/Snapshotter.kt

```diff
@@ -29,49 +34,52 @@ open class Snapshotter : MapSnapshotterObserver {
   private var snapshotCreatedCallback: SnapshotCreatedListener? = null
   private var snapshotStyleCallback: SnapshotStyleListener? = null
 
+  private val observer: Observer
+
+  private val mainHandler: Handler
+
   constructor(context: Context, options: MapSnapshotOptions) {
     this.context = WeakReference(context)
-    coreSnapshotter = MapSnapshotter(options, this)
+    coreSnapshotter = MapSnapshotter(options)
     pixelRatio = context.resources.displayMetrics.density
+    mainHandler = Handler(context.mainLooper)
+
+    observer = object : Observer() {
+      override fun notify(event: Event) {
+        when (event.type) {
+          MapEvents.MAP_LOADING_ERROR -> {
+            snapshotStyleCallback?.onDidFailLoadingStyle(event.getMapLoadingErrorEventData().message)
+            mainHandler.post {
+              coreSnapshotter.unsubscribe(this)
+            }
```

Comment:
Create a private function for that? Same as for `NativeObserver`.
Until it's not fixed upstream.